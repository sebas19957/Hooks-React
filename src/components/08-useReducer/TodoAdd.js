import React from 'react';
import { useForm } from '../../hooks/useForm';

export const TodoAdd = ({handleAddTodo}) => {

    const [{description}, handleInputChange, reset] = useForm({
        description: ''
    });

    const handleSummit = (e) => {
        e.preventDefault();

        if (description.trim() !== '') {
            const newTodo = {
                id: new Date().getTime(),
                desc: description,
                done: false
            }
            handleAddTodo(newTodo);
            reset();   
        }else{
            console.log('Campo vacio');
        }
    }

    return (
        <>
            <h4>Agregar TODO</h4>
            <hr/>

            <form>
                <input
                    type="text"
                    name="description"
                    className="form-control"
                    placeholder="Aprender ..."
                    autoComplete="off"
                    value={description}
                    onChange={handleInputChange}
                />
                <button
                    type="submit"
                    className="btn btn-outline-primary mt-2 btn-block"
                    onClick={handleSummit}
                >
                    Agregar
                </button>
            </form> 
        </>
    )
}
