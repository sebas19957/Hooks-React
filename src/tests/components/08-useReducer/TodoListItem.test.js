import { shallow } from 'enzyme';
import React from 'react';
import { TodoListItem } from '../../../components/08-useReducer/TodoListItem';
import { demoTodos } from '../../fixtures/demoTodos';

describe('Pruebas en el <TodoListItem/>', () => {

    const handleDelete = jest.fn();
    const handleToggle = jest.fn()

    const wrapper = shallow(
        <TodoListItem
            todo={demoTodos[0]}
            index={0}
            handleDelete={handleDelete}
            handleToggle={handleToggle}   
        />
    );

    test('debe de mostrarse correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe llamar la función handleDelete', () => {
        wrapper.find('button').simulate('click');
        expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
    });

    test('debe llamar la función handleToggle', () => {
        wrapper.find('p').simulate('click');
        expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
    });


    test('debe mostrar el texto correctamente', () => {
        const p = wrapper.find('p');
        expect(p.text()).toBe(`1. ${demoTodos[0].desc}`);
    });

    test('debe de tener la clase complete si el TODO.done = true', () => {

        const todo = demoTodos[0];
        todo.done = true;

        const wrapper = shallow(
            <TodoListItem
                todo={todo}
            />
        );

        expect(wrapper.find('p').hasClass('complete')).toBe(true);
    });
    
});
