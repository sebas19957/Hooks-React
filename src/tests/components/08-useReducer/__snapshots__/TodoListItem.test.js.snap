// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <TodoListItem/> debe de mostrarse correctamente  1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . 
    Aprender React
  </p>
  <button
    className="btn btn-danger ml-5"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
